sparseSpACE/HP_Optimization.py

```diff
@@ -0,0 +1,588 @@
+			return 0.0
+		params = [params[0], int(params[1]), int(params[2]), int(params[3])]
+		cur_data = self.data.copy()
+		#should implement a smoother way to put in the data set
+		classification = deml.Classification(cur_data, split_percentage=0.8, split_evenly=True, shuffle_data=False)
+		classification.perform_classification(masslumping=params[1], lambd=10**(-params[0]), minimum_level=params[2], maximum_level=self.max_lv, one_vs_others=params[3], print_metrics=False)
+		evaluation = classification.evaluate()
+		#wenn Zeit mit reingerechnet werden soll o.ä. in dieser Funktion
+		return evaluation["Percentage correct"]
+
+	def pea_classification_dimension_wise(self, params):
+		if(len(params)<4):
+			print("too little parameters for pea_classification. Returning 0.0")
+			return 0.0
+		#lambd, massl, min_lv, one_vs_others / error calc, margin(float 0-1), rebalancing (bool), use_relative_surplus (bool), max_evaluations (int)
+		params = [float(params[0]), int(params[1]), int(params[2]), int(params[3]), float(params[4]), int(params[5]), int(params[6]), int(params[7])]
+		cur_data = self.data.copy()
+		error_calculator=sparseSpACE.ErrorCalculator.ErrorCalculatorSingleDimMisclassificationGlobal()
+		ec = None
+		ovo = False
+		if(params[3] == 2):
+			ec = error_calculator
+			ovo = True
+		elif(params[3] == 1):
+			ovo = True
+		classification = deml.Classification(cur_data, split_percentage=0.8, split_evenly=True, shuffle_data=False)
+		classification.perform_classification_dimension_wise(masslumping=params[1], lambd=10**(-params[0]), minimum_level=params[2], maximum_level=self.max_lv, one_vs_others=ovo, error_calculator = ec, margin = params[4], rebalancing = params[5], use_relative_surplus = params[6], max_evaluations = params[7], print_metrics=False)
+		evaluation = classification.evaluate()
+		#wenn Zeit mit reingerechnet werden soll o.ä. in dieser Funktion
+		return evaluation["Percentage correct"]
+
+#anderes dataset ausprobieren?, vllt höhere dimensionen, bis zu dim=10. Note: moons immer 2d, aber z.B.
+def simple_test(params):
+	return params[0]
+simple_space = [["interval", 20.4, 0.]]
+
+dataset_iris = deml.datasets.load_iris(return_X_y=True)
+iris_data = deml.DataSet(dataset_iris, name="data_iris")
+OC = Optimize_Classification(data_name = "moons")
+#HPO = HP_Optimization(OC.pea_classification, OC.classification_space, r = 1.5)
+HPO = HP_Optimization(OC.pea_classification, OC.classification_space)
+#HPO = HP_Optimization(simple_test, simple_space, f_max = 100, r = 21)
+#HPO.perform_BO(5)
+#y_r = HPO.perform_RO(10)[3]
+#sol_b = HPO.perform_BO(6)
+#y_b = sol_b[3]
+#x_b = sol_b[2]
+#print("y random: " + str(y_r))
+#print("y bayesian: " + str(y_b))
+#print("x bayesian: " + str(x_b))
+#print(OC.pea_classification_dimension_wise([0.0, 0, 1, 0, 0.0, 0, 0, 2]))
+
+#HPO = HP_Optimization(simple_test, simple_space)
+#print(HPO.hp_space)
+#HPO.perform_GO(search_space=[[1, 1, 2, 1], [5, 4, 2, 0], [3, 3, 2, 1]])
+
```

Comment:
hier entfernen und evtl. in ein example aufnehmen (Das dann in einem anderen Ordner liegt oder in einem ipynb).